### PR TITLE
debugSource use uri.from

### PR DESCRIPTION
**Pull request by isidorn**
Tackles issue outlined here https://github.com/microsoft/vscode/pull/73849

Tested this a bit and does not seem to break.

### CODE REVIEW

File: src/vs/workbench/contrib/debug/common/debugSource.ts [hunk 1]
```diff
@@ -49,7 +49,11 @@ export class Source {
 		}
 
 		if (typeof this.raw.sourceReference === 'number' && this.raw.sourceReference > 0) {
-			this.uri = uri.parse(`${DEBUG_SCHEME}:${encodeURIComponent(path)}?session=${encodeURIComponent(sessionId)}&ref=${this.raw.sourceReference}`);
+			this.uri = uri.from({
+				scheme: DEBUG_SCHEME,
+				path,
+				query: `session=${encodeURIComponent(sessionId)}&ref=${this.raw.sourceReference}`
```
Comment (jrieken) [inline src/vs/workbench/contrib/debug/common/debugSource.ts]: Any special reason why you encode `sessionId`. Can this contain `=` or `&`?

Comment (isidorn) [reply src/vs/workbench/contrib/debug/common/debugSource.ts]: No, this should be a hex UUID. So I will remove the encode / decode part of the sessionID
Thanks

Comment (jrieken) [reply src/vs/workbench/contrib/debug/common/debugSource.ts]: check

**Review by jrieken — APPROVED** [review]

**Comment by isidorn** [comment]
@weinand merging this in. Let's keep our eyes open for potential regressions.

File: src/vs/workbench/contrib/debug/common/debugSource.ts (hunk 1 above)
Comment (weinand) [reply src/vs/workbench/contrib/debug/common/debugSource.ts]: @isidorn make sure to remove the corresponding decodeURIComponent call too.

Comment (isidorn) [reply src/vs/workbench/contrib/debug/common/debugSource.ts]: @weinand I just removed it. Thanks

**Pull request merged**